### DISCUSSION TITLE
how to switch relais with hardware switch on esp32

### DISCUSSION THREAD

**Discussion by dannyv1101** [arendst/Tasmota#17280] [General]
Hi, I want to build my own device whit an esp32, for that I have set up in the template:

Relais 1 on GPIO 5
Relais 2 on GPIO 12
Relais 3 on GPIO 15
Relais 4 on GPIO 18
Switch 1 on GPIO 2
Switch 2 on GPIO 4
Switch 3 on GPIO 13
Switch 4 on GPIO 14

If Switch 1 is 1 I want to turn on Relais 1, if switch 2 is 1 I want to turn on relais 2 and so on. Just switch on and Relais 1 is working, how I have to set this up?

**Comment by sfromis** [comment]
Just config the pins, default action of Tasmota is to couple switch inputs and relay outputs.

**Comment by dannyv1101** [comment]
Thank you! Now it works, I think I can not use every pin for every function.

How I have to set the thing up that when relais 1 is switching to 1 that all other relais get turn off? So that alway just one relais can be turned on.

**Reply by sfromis** [reply]
With ESP32, there are very few restrictions on which pins you can use for which function, outside of pretty specialized things like analog. To only have one relay on, you can configure the `Interlock` feature.

**Reply by dannyv1101** [reply]
Ok Thank you very much!
